packages/jam/transition/externalities/accumulate-externalities.ts

```diff
@@ -423,35 +427,68 @@ export class AccumulateExternalities
       return Result.error(NewServiceError.InsufficientFunds);
     }
 
+    // `a`: https://graypaper.fluffylabs.dev/#/ab2cdbd/366b02366d02?v=0.7.2
+    const newAccount = ServiceAccountInfo.create({
+      codeHash,
+      balance: thresholdForNew,
+      accumulateMinGas,
+      onTransferMinGas,
+      storageUtilisationBytes: bytes.value,
+      storageUtilisationCount: items,
+      gratisStorage,
+      created: this.currentTimeslot,
+      lastAccumulation: tryAsTimeSlot(0),
+      parentService: this.currentServiceId,
+    });
+
+    const newLookupItem = new LookupHistoryItem(codeHash.asOpaque(), clampedLength, tryAsLookupHistorySlots([]));
+
+    // `s`: https://graypaper.fluffylabs.dev/#/ab2cdbd/361003361003?v=0.7.2
+    const updatedCurrentAccount = ServiceAccountInfo.create({
+      ...currentService,
+      balance: tryAsU64(balanceLeftForCurrent),
+    });
+
+    if (Compatibility.isGreaterOrEqual(GpVersion.V0_7_1)) {
+      if (
+        serviceId < MIN_PUBLIC_SERVICE_INDEX &&
+        this.currentServiceId === this.updatedState.getPrivilegedServices().registrar
+      ) {
+        // NOTE: It's safe to cast to `Number` here, bcs here service ID cannot be bigger than 2**16
+        const newServiceId = tryAsServiceId(Number(serviceId));
+        if (this.getServiceInfo(newServiceId) !== null) {
+          return Result.error(NewServiceError.ServiceAlreadyExists);
+        }
+        // add the new service with selected ID
+        // https://graypaper.fluffylabs.dev/#/ab2cdbd/36be0336c003?v=0.7.2
+        this.updatedState.stateUpdate.services.servicesUpdates.push(
+          UpdateService.create({
+            serviceId: newServiceId,
+            serviceInfo: newAccount,
+            lookupHistory: newLookupItem,
+          }),
+        );
+        // update the balance of current service
+        // https://graypaper.fluffylabs.dev/#/ab2cdbd/36c20336c403?v=0.7.2
+        this.updatedState.updateServiceInfo(this.currentServiceId, updatedCurrentAccount);
+        return Result.ok(newServiceId);
+      }
```

Comment:
```suggestion
      } else {
         // in case the service is not a registrar or the requested serviceId is out of range, 
         // we completely ignore the `requestedServiceId` and assign a random one
      }
```